drivers/gles3/rasterizer_gles3.cpp

```diff
@@ -347,7 +347,8 @@ void RasterizerGLES3::blit_render_target_to_screen(RID p_render_target, const Re
 	}
 	glReadBuffer(GL_COLOR_ATTACHMENT0);
 	glBindFramebuffer(GL_DRAW_FRAMEBUFFER, RasterizerStorageGLES3::system_fbo);
-	glBlitFramebuffer(0, 0, rt->width, rt->height, p_screen_rect.position.x, win_size.height - p_screen_rect.position.y - p_screen_rect.size.height, p_screen_rect.position.x + p_screen_rect.size.width, win_size.height - p_screen_rect.position.y, GL_COLOR_BUFFER_BIT, GL_NEAREST);
+	GLenum filter = storage->texture_get_flags(rt->texture) & VS::TEXTURE_FLAG_FILTER ? GL_LINEAR : GL_NEAREST;
+	glBlitFramebuffer(0, 0, rt->width, rt->height, p_screen_rect.position.x, win_size.height - p_screen_rect.position.y - p_screen_rect.size.height, p_screen_rect.position.x + p_screen_rect.size.width, win_size.height - p_screen_rect.position.y, GL_COLOR_BUFFER_BIT, filter);
```

Comment:
When blitting a framebuffer the source and destination need to have the same MSAA setting. As far as I recall, the system_fbo always has an MSAA of 0. Therefore this will break whenever a user uses MSAA on their render target.

Reply:
Isn't this something the current code is doing wrong anyway (and is the cause of https://github.com/godotengine/godot/issues/18320)?

If so, we can merge this safely.

Reply:
@clayjohn Bump :slightly_smiling_face: 

More people have been encountering https://github.com/godotengine/godot/issues/18320 lately, so it would be good to find a fix for it in the process.